pkg/buffer_config/buffer_config.go

```diff
@@ -43,6 +43,8 @@ type DataSetBufferSettings struct {
 	RetryInitialInterval     time.Duration
 	RetryMaxInterval         time.Duration
 	RetryMaxElapsedTime      time.Duration
+	// defines timeout period (for which client will retry on failures) for processing events and sending buffers after shutdown of a client
+	RetryShutdownTimeout time.Duration
```

Comment:
What do you think about moving it out of the BufferConfig - and name it rather `gracePeriod` or something like this, since this is the wording that is used in docker-compose or k8s.

Then it would make the documentation simpler - "set grace period to the same value as you have set grace period for your container".

Reply:
Grace period is something which takes place when some time (threshold) is out. In this case its more timeout rather then grace period.

I was thinking of extracting it to client config, but decided to leave it with other retry and buffer sending configurations

Reply:
I like the idea of calling it shutdownGracePeriod on the other hand it makes thing bit more complicated since we should care of total elapsed time of shutdown not only retrials (as you stated in other comment). I would keep it more simple and let this configuration drive only retrials not whole shutdown time.